services/packagist/packagist-php-version.service.js

```diff
@@ -68,17 +68,68 @@ module.exports = class PackagistPhpVersion extends BasePackagistService {
     }
   }
 
-  transform({ json, user, repo, version = '' }) {
-    const packageVersion =
-      version === ''
-        ? this.getDefaultBranch(json, user, repo)
-        : json.packages[this.getPackageName(user, repo)][version]
+  findVersionIndex(json, version) {
+    return json.findIndex(v => v.version === version)
+  }
+
+  async findSpecifiedVersion(json, user, repo, version, server) {
+    let release
+
+    if ((release = json[this.findVersionIndex(json, version)])) {
+      return release
+    } else {
+      try {
+        const allData = await this.fetchDev({
+          user,
+          repo,
+          schema: allVersionsSchema,
+          server,
+        })
+
+        const decompressed = this.decompressResponse(
+          allData,
+          this.getPackageName(user, repo)
+        )
+
+        return decompressed[this.findVersionIndex(decompressed, version)]
+      } catch (e) {
+        return release
+      }
+    }
+  }
+
+  async transform({ json, user, repo, version = '', server }) {
+    let packageVersion
+    const decompressed = this.decompressResponse(
+      json,
+      this.getPackageName(user, repo)
+    )
+
+    if (version === '') {
+      packageVersion = this.findRelease(decompressed)
+    } else {
+      try {
+        packageVersion = await this.findSpecifiedVersion(
+          decompressed,
+          user,
+          repo,
+          version,
+          server
+        )
+      } catch {
```

Comment:
stylistically we always use `} catch (e) {`
we should probably have an eslint rule for that